### PR TITLE
[443] ETH/WETH Wrapping logic

### PR DESCRIPTION
**Pull request by W3stside**
Adds ETH/WETH wrapping logic into #443 

Waterfalls into #449 

1. Shows loader and disables `Wrap my NATIVE_TOKEN` button on action - wraps the inputAmount of "from" token
2. menu urls are external target="_blank" behaviour (did this here because ease and not a huge app breaking change)
3. passes an optional param to `useWrapCallback` called `isEthTradeOverride` which is input to signal to app that we are selling ETH and the output token is NOT WETH - this disables blocking the app to show the normal ETH/WETH wrap flow

### QA TESTING
1. all steps from QA testing in #449 
2. Wrap my NATIVE_TOKEN should prompt MM/wallet and wrapping should work with:
  - wallet modal pops up
  - button shows loading spinner and is disabled (can't be clicked why tx in progress)
  - Pending transactions reacts to change and shows transaction properly
  - when confirmed balances update and loading goes away

### Notes
1. when exiting swap page and going back state is lost in trade wiping all wrap ETH to WETH visualisations. Pending transactions remain, and the tx will no fail as it's already been sent to BC, but UX could be weird.

### CODE REVIEW

**Review by alfetopito — COMMENTED** [review]
I worked great, managed to wrap my eth without issues.

One point to note though, is that the `wrap` button has the loading indicator and is disabled only while the wallet modal is open.
Once the tx is sent, the button is enabled and I can click again and send another tx while the first one is still pending.

I understood from the description that this shouldn't be the case

File: src/custom/components/Menu/index.tsx [hunk 1]
```diff
@@ -14,7 +14,10 @@ export const StyledMenu = styled(MenuMod)`
   }
 `
 
-const Policy = styled(InternalMenuItem)`
+const Policy = styled(InternalMenuItem).attrs(attrs => ({
+  ...attrs,
+  target: '_blank'
+}))`
```
Comment (anxolin) [inline src/custom/components/Menu/index.tsx:20]: Yep, I red the comment, but problem is that this unrelated change might add additional unrelated discussion. 

For example, in this case, I don't think we should add this. Why do we want to open the policies in a blank page? Is this a requirement coming from somewhere?

File: src/custom/components/swap/EthWethWrap/index.tsx [hunk 2]
```diff
@@ -28,11 +29,16 @@ const Wrapper = styled.div`
   }
 
   > ${ButtonPrimary} {
-      // TODO: themed
+      // TODO: @biocom themed
```
Comment (anxolin) [inline src/custom/components/swap/EthWethWrap/index.tsx:32]: Maybe is best an issue, so @biocom can add it to his backlog?

File: src/custom/components/swap/EthWethWrap/index.tsx [hunk 3]
```diff
@@ -79,13 +85,31 @@ export interface Props {
...
+      setLoading(true)
+      setError(null)
+
+      await wrapCallback()
+    } catch (error) {
+      console.error('Error wrapping ETH:', error)
+
+      setError(new Error(error))
+    } finally {
+      setLoading(false)
```
Comment (anxolin) [inline src/custom/components/swap/EthWethWrap/index.tsx:110]: I see how this is simpler than the original approach we discussed. I think we can do it in a separate PR the part where if there's at least one WRAPING going on, we are "loading".

**Review by anxolin — APPROVED** [review]
Great job! 👌

Same comment as Leandro. But I think we can do in another PR. 
Also the `_blank` thing. Approving, but I think we should revert that imo.

**Comment by anxolin** [comment]
@W3stside happy to share with QA this, but I think we still have some UX changes we will include. I wonder if we want to do the testing now, or once the feature is 100% complete: warnings, improved layout, loading until the tx is mined, etc.

If you agree, we would share with @MareenG once we freeze version, or at least when all the missing bits are ready.

File: src/custom/components/Menu/index.tsx (hunk 1 above)
Comment (alfetopito) [reply src/custom/components/Menu/index.tsx:20]: David mentioned this somewhere (don't remember where). To not loose state/avoid over complicate state handling.
If that's loaded in the same page while one action is pending, the state will be lost.

Comment (anxolin) [reply src/custom/components/Menu/index.tsx:20]: Mm, i don't understand. It feels strange to me to open a _blank page to the same app. 
If they navigate, its fine to lose the state

I'm probably missing sth

**Review by alfetopito — APPROVED** [review]

File: src/custom/components/Menu/index.tsx (hunk 1 above)
Comment (biocom) [reply src/custom/components/Menu/index.tsx:20]: I think as David points out, between the trade-off of keeping state vs. navigating (and priorities..) I think a target=_blank is an option for now. Just make sure to do the `target="_blank" rel="noopener"` for security.

File: src/custom/components/swap/EthWethWrap/index.tsx (hunk 3 above)
Comment (W3stside) [reply src/custom/components/swap/EthWethWrap/index.tsx:110]: don't understand this comment but let's discuss everything in another PR, merging this to consolidate

File: src/custom/components/Menu/index.tsx (hunk 1 above)
Comment (W3stside) [reply src/custom/components/Menu/index.tsx:20]: let's discuss this in subsequent PR(s)
